morf-oracle/src/main/java/org/alfasoftware/morf/jdbc/oracle/OracleDialect.java

```diff
@@ -268,6 +269,26 @@ public Collection<String> dropStatements(Table table) {
     return Arrays.asList("DROP TABLE " + schemaNamePrefix() + table.getName());
   }
 
+  @Override
+  public Collection<String> dropTables(List<Table> tables, boolean ifExists, boolean cascade) {
+    String tablesString = tables.stream().map(s -> "'" + s.getName().toUpperCase() + "'").collect(Collectors.joining(", "));
+    StringBuilder sb = new StringBuilder();
+    sb.append("BEGIN\n");
+    sb.append("  FOR T IN (\n");
+    sb.append("    SELECT '").append(schemaNamePrefix()).append("' || TABLE_NAME AS TABLE_NAME\n");
+    if (ifExists) {
+      sb.append("    FROM ALL_TABLES\n");
+      sb.append("   WHERE TABLE_NAME  IN (").append(tablesString).append(")\n");
+    } else {
+      sb.append("    FROM (SELECT COLUMN_VALUE AS TABLE_NAME from TABLE(SYS.dbms_debug_vc2coll(").append(tablesString).append(")))\n");
+    }
+    sb.append("  )\n");
+    sb.append("  LOOP\n");
+    sb.append("    EXECUTE IMMEDIATE 'DROP TABLE ' || T.TABLE_NAME ").append(cascade ? "|| ' CASCADE CONSTRAINTS ' " : "").append(";\n");
+    sb.append("  END LOOP;\n");
+    sb.append("END;");

```

Comment:
@tkominiak - If we're running 100+ drop statements with ifExists = false, will we see a performance hit using this approach versus a single DROP statement? With this approach we're always going to do a table read before we drop, even if we haven't specified that we want an if exists check.